crates/db_schema/src/impls/post.rs

```diff
@@ -96,6 +103,16 @@ impl Post {
       .await
   }
 
+  pub async fn list_for_sitemap(pool: &mut DbPool<'_>, limit: i64) -> Result<Vec<Self>, Error> {
+    let conn = &mut get_conn(pool).await?;
+    post
+      .filter(deleted.eq(false).and(removed.eq(false)))
+      .order(published.desc())
+      .limit(limit)
+      .load::<Self>(conn)
+      .await
+  }
```

Comment:
It also needs the `published` or `updated` field for `<lastmod>` in the sitemap.

I have also updated the query to select the `updated` field and select `published` if it is `None`. The recommendation for the `<lastmod>` element is to set it to 'the last time the content of the page changed meaningfully' https://developers.google.com/search/blog/2014/10/best-practices-for-xml-sitemaps-rssatom#last-modification-time

I'm not sure if this is the best solution here, since 'changed meaningfully' isn't a very precise definition.

Reply:
This should probably only select posts in local communities.

Also 50000 is way too much. I would rather filter for posts which were created within the last 24 hours.

Reply:
@Nutomic I agree on the local posts, that makes a lot more sense.

I was thinking that 50000 is quite a lot, too, but thought it would be ok since we're only running it once a day, because of the cache policy. I've been looking into sitemap best practices and most people say the sitemap should ideally contain ALL of the content on your page. I've not really found any good information if a "partial" sitemap is problematic.
From what I understand currently, the sitemap only seems to signal to web crawlers which of your sites are the most relevant ones for indexing. I don't think removing a url from the sitemap (i.e it's older than 24 hours and therefore not included in the query) will make a web crawler stop indexing it, but I also don't know what effects it might have. 

I'm not really a SEO person, so maybe someone with more insight into it could give some feedback, also in regards to the relatively long sitemap cache of 1 day.

Reply:
Not every instance is going to configure nginx caching, so for those it will hit the db every time the sitemap is retrieved. And to my understanding the sitemap is basically a list of urls which search engines should crawl. They will also crawl other urls if they find them, and definitely wont remove working urls from the index just because they arent in the site map. So what you have now looks good to me, any reason its still marked as draft?

Reply:
The only reason it's still a draft is because I got sidetracked by other stuff 😄 I will convert it later today. There is one outstanding issue regarding the timezone offset, so maybe this PR should be blocked until #3496 is merged?

Reply:
Probably shouldn't be a blocker, and the TODO is there in the code. You can also open up an issue to remember if you like.